tests/ut/ir/expressions/test_call_arg_directions.py

```
@@ -0,0 +1,242 @@
+        call = ir.Call(op, [x, y, z], dirs, {}, _make_type(), _span())
+        assert [d for d in call.arg_directions] == dirs
+
+    def test_explicit_empty_directions_allowed(self):
+        # An explicit empty list is equivalent to "legacy / not yet derived".
+        op = _make_op()
+        call = ir.Call(op, [_scalar_var("x")], [], {}, _make_type(), _span())
+        assert list(call.arg_directions) == []
+
+    def test_size_mismatch_raises(self):
+        op = _make_op()
+        with pytest.raises(Exception):  # noqa: PT011  TypeError raised from C++
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)